pyensign/ensign.py

```diff
@@ -81,11 +95,11 @@ async def publish(self, topic_name, *events):
         if len(events) == 0:
             raise ValueError("no events provided")
 
-        # Get the topic ID from the topic name
-        topic_id = await self.topic_id(topic_name)
-
-        # Ensure the topic exists
-        if topic_id == "":
+        # Ensure the topic ID exists by getting or creating it
+        topic_id = ""
+        try:
+            topic_id = await self.topic_id(topic_name)
+        except EnsignTopicNotFoundError:
```

Comment:
I suppose we might have other possible errors here in the future -- I heard that there's a new PEP for something called "Exception Groups" which might be neat to look at! https://peps.python.org/pep-0654/

Reply:
Right, exception handling is still kind of an open question for the client methods, especially since they are async. I think we need more experience using the SDK to see how errors are manifested for the users, and I expect that the exception group will come in handy eventually.